docs/reference/builder.md

```diff
@@ -226,8 +248,61 @@ following lines are all treated identically:
 
 The following parser directive is supported:
 
+* `syntax`
 * `escape`
 
+## syntax
+
+    # syntax=[remote image reference]
+
+For example:
+
+    # syntax=docker/dockerfile
+    # syntax=docker/dockerfile:1.0
+    # syntax=docker/dockerfile:1.0.0-experimental
```

Comment:
should we include an example with a fully-qualified reference for Docker Hub as well? (`docker.io/.....`)